### ISSUE TITLE
Warning message about numpy.sqrt seen when running sherpa_test

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#30]
It appears that, if you have numpy 1.9.2 installed, `sherpa_test` will produce the following on-screen warning (originally discussed as part of PR #29 but has been pulled out into a separate issue). I don't think anything is wrong here, but

 - it's visually ugly
 - I can't explain the change in behavior, which makes me nervous

```
.................................ssssssssssssssssssssssssssss.......sss..ssssssssssss./home/naridge/data/my-sherpa/sherpa-omar/sherpa/estmethods/__init__.py:369: RuntimeWarning: invalid value encountered in sqrt
  diag = (sigma * numpy.sqrt(inv_info)).diagonal()
....sssssssssss..........................................................ssssss...........................sssss.sssssss.......................................................
----------------------------------------------------------------------
```

This does not happen if numpy 1.8.1 is installed.

Running `sherpa_test -v 3` shows that the message is from `test_covar`

```
test_covar (sherpa.estmethods.tests.test_estmethods.test_estmethods)/home/naridge/data/my-sherpa/sherpa-omar/sherpa/estmethods/__init__.py:369: RuntimeWarning: invalid value encountered in sqrt
  diag = (sigma * numpy.sqrt(inv_info)).diagonal()
 ... ok
```

However, I get this warning from both versions of numpy when running `sqrt` directly, as shown below, so it's not clear to me what is causing this message to appear. A quick look at `test_estmethods.py` doesn't suggest that the warning is being "turned off" in any way - I haven't looked to see if `numpy.seterr` allows this - but maybe there's some change in behavior in one of the earlier tests that means that with numpy 1.9 the message hasn't already been created, whereas it is with numpy 1.8 (in a place where the message is "hidden"); this is all speculation.

```
% python
Python 2.7.9 |Continuum Analytics, Inc.| (default, Apr 14 2015, 12:54:25) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://binstar.org
>>> import numpy as np
>>> np.version.version
'1.9.2'
>>> np.sqrt([1,2,3])
array([ 1.        ,  1.41421356,  1.73205081])
>>> np.sqrt([1,-2,3])
__main__:1: RuntimeWarning: invalid value encountered in sqrt
array([ 1.        ,         nan,  1.73205081])
>>> np.sqrt([1,-2,3])
array([ 1.        ,         nan,  1.73205081])
>>> 
```

and

```
% python
Python 2.7.6 (default, Apr 18 2014, 17:48:56) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.version.version
'1.8.1'
>>> np.sqrt([1,2,3])
array([ 1.        ,  1.41421356,  1.73205081])
>>> np.sqrt([1,-2,3])
__main__:1: RuntimeWarning: invalid value encountered in sqrt
array([ 1.        ,         nan,  1.73205081])
>>> np.sqrt([1,-2,3])
array([ 1.        ,         nan,  1.73205081])
>>> 
```

**Comment by DougBurke** [comment]
@olaurino can I run tests directly/individually?

**Comment by DougBurke** [comment]
Aha, when I have numpy 1.8.2 installed, and print out the contents of `numpy.geterr()['invalid']` just before the call to the `numpy.sqrt` call, the value is `ignore` but with 1.9.1 it's `warn`. So this explains the difference in behaviour.  However, something in the code must be changing this, as  the default value is `warn`, even in 1.8.2 (shown below). I wonder if there's a slightly different code path here (perhaps because `pyfits_backend` is loaded in one but not the other...)?

In [1]: import numpy

In [2]: numpy.version.version
Out[2]: '1.8.2'

In [3]: numpy.geterr()['invalid']
Out[3]: 'warn'
```

**Comment by DougBurke** [comment]
I don't seem to be seeing this behaviour in PR #31.

**Comment by dtnguyen2** [comment]
I can't seem to reproduce this warning message.  I have tried various combinations of numpy versions and still no warning messages, there must be other combo which triggers this.

**Comment by DougBurke** [comment]
@dtnguyen2 - I think you need to be using an old version of the repository, before #31 was merged - e.g. e59b455

**Comment by dtnguyen2** [comment]
I think I have checked out the correct version, and still I don't see any
warning message.

(sherpa-dev)[dtn@newdevel12 sherpa]$ git log --oneline
--first-parent
0daa0d2 Merge PR #28 (taldcroft) - skip tests instead of pretending to
pass
ec080fe Merge PR #18 (olaurino) - improve .gitignore and remove generated
region files
342b7ea Merge PR #23 (olaurino) - make python setup.py develop transparent
to user (fix #14)
e66c856 Merge PR #21 (olaurino) - Upgrade versioneer (fix
#13)
872d834 Merge PR #17 (DougBurke) - docstring updates (ahelp
ingest)
90680f0 Merge pull request #7 from
taldcroft/gitignore
4d23439 Merge branch
'release/4.7'
ba26a5f fix version in README.md
instructions
e623675 change conda channel to
sherpa
ca3ec8a update conda
recipe
2a1e54c add disclarimer to
README.md
da1f8d0 add licensing and copyright
information
9ea2931 add basic documentation: readme and quick start
notebook
cd773e8 fix numpytest copyright header and remove non-ascii
characters
dd46d7b add sherpa-standalone.rc
support
61fbfcd fix None comparisons triggering warnings in numpy
1.9
428b216 revert iter to
3
e1ed841 apply changes from
/main/LATEST
6c6f1dd make test_sim fail gracefully when xspec is not
present
53e26c0 get rid of warnings from numpy 1.8 to 1.9 OTS
upgrade
ac5b8dd remove unused powell, odrpack, port
code
b25fffa remove unused
StreamingException
1a65089 remove pykdtree, its dependencies, and
tests
29ac8fc add --enable-standalone flag to config (new region
lib)
39a0902 rename grplib
folder
1fd7c09 upgrade region and group
code
cb974ad add more copyright headers to file that were missing
one
835c6e7 replace copyright
macros
111768e reinstate non-CIAO version string for
git
aa3dbfb fix sherpa.__version__ for
4.7
ff1b07f update conda
recipe/meta
d683dac Load template models in session's
namespace
af79955 fix #14044: plots in datastack - confused
windows?
aba2273 workaround distutils issue #19286 with different package_data
pattern
69a7d54 avoid Python.h override
warning
9c8837d some build fixes (ax_python_devel,
distutils)
7a7f9b9 fix
#14001

(sherpa-dev)[dtn@newdevel12 sherpa]$ git checkout
0daa0d2
M
setup.cfg

Note: checking out
'0daa0d2'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at 0daa0d2... Merge PR #28 (taldcroft) - skip tests instead of
pretending to pass
(sherpa-dev)[dtn@newdevel12
sherpa]$
(sherpa-dev)[dtn@newdevel12 sherpa]$
sherpa_test
  Found 7/7 tests for
sherpa.astro.tests.test_datastack
  Found 1/1 tests for
sherpa.astro.models.tests.test_models
  Found 25/25 tests for
sherpa.astro.tests.test_optical
  Found 2/2 tests for
sherpa.astro.sim.tests.test_sim
  Found 26/26 tests for
sherpa.astro.tests.test_astro
  Found 6/6 tests for sherpa.astro.tests.test_data
  Found 1/1 tests for sherpa.astro.tests.test_plot
  Found 1/1 tests for sherpa.astro.ui.tests.test_nan
  Found 15/15 tests for sherpa.astro.ui.tests.test_ui
  Found 1/1 tests for sherpa.astro.utils.tests.test_utils
  Found 5/5 tests for sherpa.astro.tests.test_xspec
  Found 4/4 tests for sherpa.estmethods.tests.test_estmethods
  Found 6/6 tests for sherpa.image.tests.test_image
  Found 1/1 tests for sherpa.models.tests.test_basic
  Found 14/14 tests for sherpa.models.tests.test_model
  Found 12/12 tests for sherpa.models.tests.test_parameter
  Found 6/6 tests for sherpa.models.tests.test_template
  Found 26/26 tests for sherpa.optmethods.tests.test_optmethods
  Found 18/18 tests for sherpa.plot.tests.test_plot
  Found 18/18 tests for sherpa.sim.tests.test_sim
  Found 5/5 tests for sherpa.tests.test_sherpa
  Found 1/1 tests for sherpa.tests.test_stack
  Found 8/8 tests for sherpa.tests.test_ui
  Found 1/1 tests for sherpa.utils.tests.test_err
  Found 1/1 tests for sherpa.utils.tests.test_integration
  Found 43/43 tests for sherpa.utils.tests.test_root
  Found 10/10 tests for sherpa.utils.tests.test_utils
.................................ssssssssssssssssssssssssssss.......ssss..ssssssssss...sERROR:
XSXSapec model evaluation failed
F......................................sssss...............................ssssss..........................sssss..sssssss.......................................................
======================================================================
FAIL: test_xspec_models (sherpa.astro.tests.test_xspec.test_xspec)
----------------------------------------------------------------------
Traceback (most recent call last):
  File
"/data/scialg/staff/dtn/src/gitsherpa/sherpa/sherpa/astro/xspec/tests/test_xspec.py",
line 87, in test_xspec_models
    not numpy.isinf(vals).any() )
AssertionError: False is not true

----------------------------------------------------------------------
Ran 264 tests in 33.681s

FAILED (failures=1, skipped=66)

**Comment by DougBurke** [comment]
I am closing this out as its no-longer reproducible and changes have been made to how the tests do logging.
